### ISSUE TITLE
Save Multiple Related Records from tabular input

### ISSUE THREAD

**Issue by joshuatam** [yiisoft/yii2#6326]
Just a suggestion, when saving multiple relational records from a tabular input,
The following code may be useful:

``` php
class ExtendedActiveRecord extends ActiveRecord
{
    /**
     * Save Records for Related Models (Single Column Primary Key)
     * @param $name string
     * @param $data mixed
     */
    public function saveRelated($name, $data) {
        $transaction = Yii::$app->db->beginTransaction();

        $this->unlinkAll($name, true);

        $relation = $this->getRelation($name);
        if ($relation->via != null) {
            /** @var ActiveRecord $className */
            $className = $relation->modelClass;
            $relatedRecords = $className::findAll([$className::primaryKey()[0] => $data[$this->formName()][$name]]);
            foreach ($relatedRecords as $record) {
                $this->link($name, $record);
            }
        }

        $transaction->commit();
    }
} 
```

Usage:

Add the following line

``` php
$model->saveRelated('<relation_name>', Yii::$app->request->post());
```

after

``` php
if ($model->load(Yii::$app->request->post()) && $model->save())
```

**Comment by mootensai** [comment]
https://github.com/yiisoft/yii2/pull/2070#issuecomment-106140033

https://github.com/mootensai/yii2-relation-trait

**Comment by dynasource** [comment]
closed in favor of #2070
